cannot find type {RArray,RString} in this scope

Packaging rb-sys for Debian has shown me this specific error in multiple test cases.
`cargo test --all-targets --features bindgen-impl-debug,link-ruby` (with or without link-ruby) can reproduce this.
Not sharing the whole stacktrace as that is huge, but I'll add the relevant errors.
```bash
elp: consider importing one of these items
  |
1 | # [doc = r" Contains all items that are not yet categorized by ABI stability."] # [doc = r" These items are candidates for promotion to `stable` or `unstable`"] # [doc = r" in the future."] pub mod uncategorized { use crate::RString;
  |                                                                                                                                                                                                                       +++++++++++++++++++
1 | # [doc = r" Contains all items that are not yet categorized by ABI stability."] # [doc = r" These items are candidates for promotion to `stable` or `unstable`"] # [doc = r" in the future."] pub mod uncategorized { use crate::internal::RString;
  |                                                                                                                                                                                                                       +++++++++++++++++++++++++++++
1 | # [doc = r" Contains all items that are not yet categorized by ABI stability."] # [doc = r" These items are candidates for promotion to `stable` or `unstable`"] # [doc = r" in the future."] pub mod uncategorized { use crate::stable::RString;
  |                                                                                                                                                                                                                       +++++++++++++++++++++++++++

error[E0412]: cannot find type `RArray` in this scope
 --> /home/weepingclown/debian/rust/debcargo-conf/build/rb-sys/target/debug/build/rb-sys-83031d16783079ea/out/bindings-0.9.94-x86_64-linux-gnu-3.1.2.rs:1:86178
  |
1 | ... :: Result { write ! (f , "RArray__bindgen_ty_1 {{ union }}") } } impl :: std :: fmt :: Debug for RArray { fn fmt (& self , f : & mut :: std :: fmt :: Formatter < '_ >) -> :: std :: fmt :: Result { write ! (f , "RArray {{ basic...
  |                                                                                                      ^^^^^^ not found in this scope
  |
help: consider importing one of these items
  |
1 | # [doc = r" Contains all items that are not yet categorized by ABI stability."] # [doc = r" These items are candidates for promotion to `stable` or `unstable`"] # [doc = r" in the future."] pub mod uncategorized { use crate::RArray;
  |                                                                                                                                                                                                                       ++++++++++++++++++
1 | # [doc = r" Contains all items that are not yet categorized by ABI stability."] # [doc = r" These items are candidates for promotion to `stable` or `unstable`"] # [doc = r" in the future."] pub mod uncategorized { use crate::internal::RArray;
  |                                                                                                                                                                                                                       ++++++++++++++++++++++++++++
1 | # [doc = r" Contains all items that are not yet categorized by ABI stability."] # [doc = r" These items are candidates for promotion to `stable` or `unstable`"] # [doc = r" in the future."] pub mod uncategorized { use crate::stable::RArray;
  |                                                                                                                                                                                                                       ++++++++++++++++++++++++++

For more information about this error, try `rustc --explain E0412`.
error: could not compile `rb-sys` due to 2 previous errors

```

--- a/.github/workflows/integration.yml
+++ b/.github/workflows/integration.yml
@@ -27,26 +27,11 @@ jobs:
             run: cargo test
           - name: "matsadler/magnus"
             slug: magnus-head
-            ref: "05359fcd2369f014570f1e417c7c2d462625af5e"
+            ref: "627755429885eb1a84929e4d876f8cd8ec7303cf"
             run: cargo test
         os: ["ubuntu-latest", "windows-latest", "macos-latest"]
         rust: ["stable"]
         ruby: ["2.7", "3.2", "3.3"]
-        exclude:
-          - os: windows-latest
-            ruby: "3.3" # remove this once 3.3 binaries are available for windows
-            repo:
-              name: "matsadler/magnus"
-              slug: magnus-head
-              ref: "05359fcd2369f014570f1e417c7c2d462625af5e"
-              run: cargo test
-          - os: windows-latest
-            ruby: "3.3" # remove this once 3.3 binaries are available for windows
-            repo:
-              name: "matsadler/magnus"
-              slug: magnus-0.5
-              ref: "0.5.5"
-              run: cargo test
 
     runs-on: ${{ matrix.os }}
     steps:

--- a/Rakefile
+++ b/Rakefile
@@ -30,6 +30,7 @@ namespace :test do
   cargo_test_task "rb-sys-tests"
   cargo_test_task "rb-sys-env"
   cargo_test_task "rb-sys-test-helpers"
+  cargo_test_task "rb-sys", "--all-targets", "--features", "bindgen-impl-debug,link-ruby" # Cover debian use case
 
   desc "Test against all installed Rubies"
   task :rubies do

--- a/crates/rb-sys-build/src/bindings/stable_api.rs
+++ b/crates/rb-sys-build/src/bindings/stable_api.rs
@@ -122,6 +122,22 @@ pub fn categorize_bindings(syntax: &mut syn::File) {
             excluded_items.push(item);
         }
     }
+
+    // Perform a pass on all std::fmt::Debug implementations to fully qualify opaque structs
+    for item in excluded_items.iter_mut() {
+        if let syn::Item::Impl(ref mut impl_item) = item {
+            if let Some((_, syn::Path { segments, .. }, _)) = impl_item.trait_.as_ref() {
+                if segments.iter().any(|segment| segment.ident == "Debug") {
+                    if let syn::Type::Path(ref mut path) = *impl_item.self_ty {
+                        if opaque_idents_to_swap.contains(&path.to_token_stream().to_string()) {
+                            *impl_item.self_ty = syn::parse_quote! { crate::internal::#path };
+                        }
+                    }
+                }
+            }
+        }
+    }
+
     *syntax = syn::parse_quote! {
         /// Contains all items that are not yet categorized by ABI stability.
         /// These items are candidates for promotion to `stable` or `unstable`

